src/dbnode/persist/types.go

```diff
@@ -204,3 +204,29 @@ const (
 	// FileSetIndexContentType indicates that the fileset files contain time series index metadata
 	FileSetIndexContentType
 )
+
+// OnFlushSeries performs work on a per series level.
+type OnFlushSeries interface {
+	OnFlushNewSeries(
+		shard uint32,
+		blockStart time.Time,
+		id ident.ID,
+		tags ident.Tags,
+	) error
+}
+
+// NoOpColdFlushNamespace is a no-op impl of OnFlushSeries.
+type NoOpColdFlushNamespace struct{}
```

Comment:
Looks good, want to add enforcement fits the interface just above? `var _ OnFlushSeries = (*NoOpColdFlushNamespace)(nil)`